src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs

```diff
@@ -37,13 +41,16 @@ class KeyValuePairSettings : ILoggerSettings
         const string MinimumLevelOverrideDirectivePrefix = "minimum-level:override:";
 
         const string CallableDirectiveRegex = @"^(?<directive>audit-to|write-to|enrich|filter):(?<method>[A-Za-z0-9]*)(\.(?<argument>[A-Za-z0-9]*)){0,1}$";
+        const string LevelSwitchDeclarationDirectiveRegex = @"^level-switch:(?<switchName>\$[A-Za-z0-9]+)$";
```

Comment:
Instead of enforcing the presence of `$` here, would it be friendlier to accept anything, but emit a warning or throw when the directive is found but the name is invalid?

Reply:
Yes that would be friendlier indeed :)
I would say "throw" in here, the same way that right now for instance setting an invalid url in the Seq Sink throws a FormatException somewhere down the line.

I think throwing a `FormatException` with a detailed message would make sense. I don't think creating a custom exception type would add any value.

Reply:
Agreed :+1: